### ISSUE TITLE
Slots rendering collections issue with story validation

### ISSUE THREAD

**Issue by HermantNET** [jonspalmer/view_component-storybook#123]
Hello, I'm newish to ruby so sorry that I'm unable to provide written tests, trust me I tried lol. Anyways, the issue lies with the `rake view_component_storybook:write_stories_json` command. For context, ViewComponent allows us to render a collection like so:

```rb
# navigation_component.rb
class NavigationComponent < ViewComponent::Base
  renders_many :links, "LinkComponent"

  class LinkComponent < ViewComponent::Base
    def initialize(name:, href:)
      @name = name
      @href = href
    end
  end
end
```
```erb
<%# navigation_component.html.erb %>
<% links.each do |link| %>
  <%= link %>
<% end %>
```
```erb
<%# index.html.erb %>
<%= render(NavigationComponent.new) do |c| %>
  <% c.links([
    { name: "Home", href: "/" },
    { name: "Pricing", href: "/pricing" },
    { name: "Sign Up", href: "/sign-up" },
  ]) %>
<% end %>
```

The focus is on `c.links([...])`. Using this method to set the slot doesn't appear to be covered by the slot method validation logic. When trying to use `rake view_component_storybook:write_stories_json` the command outputs:

```
StandardError: NavigationComponentStories invalid: (Story configs 'default' is invalid: (Slots is invalid))
```

I forked the repo and disable the slots validation bit to make sure that the problem is with that bit of code:

```rb
# lib/view_component/storybook/slots/slot_config.rb

def validate_slot_method_args
  true
end
```

Sure enough everything worked as expected apart from the now lack of slot validation. I tried to fix the problem but the validation code looks like black magic and I don't understand how to write the tests so sorry I can't provide much else other than notifying you to the issue :(

I did try to write a test along the lines of `described_class.from_component(SlotsV2Component, :items, [{}, {}, {}])`, but that test passed so I'm probably missing something.

**Comment by jonspalmer** [comment]
Can you share your stories code that you are using around the component with a collection?

**Comment by jonspalmer** [comment]
I expect API changes and simplification coming in v1.0.0, specifically around slots, will eliminate this problem

**Comment by jonspalmer** [comment]
fixed in v1.0.0
